odxtools/diaglayer.py

```diff
@@ -827,7 +906,9 @@ def get_doip_routing_activation_type(self,
 
         return int(result)
 
-    def get_tester_present_time(self, protocol_name: Optional[str] = None) -> Optional[float]:
+    def get_tester_present_time(self,
+                                protocol: Optional[Union[str,
+                                                         "DiagLayer"]] = None) -> Optional[float]:
```

Comment:
The more I look at the current API, the more I think something like {{diag_layer.com(protocol).tester_present_time}} would make more sense, where `com` is a function that returns an object that encapsulate all the CP lookup logic.

Reply:
such an approach makes implementing fallbacks more complicated (i.e., "find a COMPARAM for a given protocol, if none exits, look up for one which does not specify a protocol"). It might be a good idea to put these methods into a separate "comparam lookup convenience" class, but IMO that's a topic for a separate PR...